distmv/studentst.go

```diff
@@ -0,0 +1,335 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distmv
+
+import (
+	"math"
+	"math/rand"
+	"sync"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/matrix/mat64"
+	"github.com/gonum/stat/distuv"
+)
+
+// StudentT is a multi-variate StudentT distribution. It is a distribution over
+// ℝ^n with the probability density
+//  p(y) = (Γ((ν+d)/2) / Γ(ν/2)) * (νπ)^(-n/2) * |Ʃ|^(-1/2) *
+//             (1 + 1/ν * (y-μ)^T * Ʃ^-1 * (y-μ))^(-(ν+n)/2)
+// where ν is a scalar greater than 2, μ is a vector in ℝ^n, and Ʃ is an n×n
+// symmetric positive definite matrix.
+//
+// In this distribution, ν sets the spread of the distribution, similar to
+// the degrees of freedom in a univariate Student's T distribution. As ν → ∞,
+// the distribution approaches a multi-variate normal distribution.
+// μ is the mean of the distribution, and the covariance is  ν/(ν-2)*Ʃ.
+//
+// See http://users.isy.liu.se/en/rt/roth/student.pdf for more information.
+type StudentsT struct {
+	nu  float64
+	mu  []float64
+	src *rand.Rand
+
+	once  sync.Once
+	sigma *mat64.SymDense // only stored if needed
+
+	chol       mat64.Cholesky
+	lower      mat64.TriDense
+	logSqrtDet float64
+	dim        int
+}
+
+// NewStudentsT creates a new StudentsT with the given scale, mean, and covariance
```

Comment:
Use the parameter names as they exist in the signature, or explain the relationship here between the scale, mean, and covariance and the parameters in the signature.